src/components/PayPalScriptProvider.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
```

Comment:
Let's import these in so we don't have to prefix with `React.`
```suggestion
import React, { useEffect, useReducer, FC } from "react";
```

Reply:
I'll need to spend some time here, I facing problems with that suggestion. If I remove the React I have problem with lint and typescript. posible solutions are: in .eslintrc.json under rules put this: "react/react-in-jsx-scope": "off" and in .tsconfig.json change compilerOptions.jsx to "react-jsx". But after do that all brakes and I can not build any more the project. here an example of the error:

![Screen Shot 2021-07-30 at 5 23 24 PM](https://user-images.githubusercontent.com/26287838/127717352-92dbf517-c7c1-489a-bd18-6c9d9e7cff4f.png)

Reply:
Never mind missing extension in nodeResolve() in roolup.config.js. Right config is:
`nodeResolve({"extensions": ".jsx"})`

Reply:
I think there was some confusion here. Sorry about that. We want to keep React as an import. I would like to revert the build changes and import React in along with the other things from React like so:

```js
import React, {
    useEffect,
    useRef,
    useState,
    FunctionComponent,
    ReactElement,
} from "react";
```

Reply:
side note if we update react and the tooling around it we won't need the `React` import anymore. Let's peek into that soon :D

Reply:
I already solve the problems with the React import. I can test it again and check is something is missing. I'm able to build and run test in the project after removing the React and start using the jsx transform after react version 17.

Reply:
Can we update the package to only allows REact version >=17?

Reply:
I prefer to go with the new React transform jsx. I can make a suggestion this is my principal reason:

![Screen Shot 2021-07-31 at 5 46 09 PM](https://user-images.githubusercontent.com/26287838/127754093-b920142d-ef7c-41f7-b881-78fb6c296224.png)

![Screen Shot 2021-07-31 at 5 46 27 PM](https://user-images.githubusercontent.com/26287838/127754097-ecdf853a-975b-41e5-ab5b-09763a77ce8e.png)

The size in the imports

Reply:
> I already solve the problems with the React import. I can test it again and check is something is missing. I'm able to build and run test in the project after removing the React and start using the jsx transform after react version 17.

I see two problems when testing locally:

1. VS Code shows type errors in index.ts. There's a VS Code bug related to react-jsx showing these as type errors. There are not actually type errors when running `npm run typecheck` but it's still confusing and we should avoid for now.

<img width="1460" alt="Screen Shot 2021-08-01 at 11 34 28 AM" src="https://user-images.githubusercontent.com/534034/127778575-91e960b8-5e53-420f-b350-1504b59c8621.png">

2. The task `npm run build` throws a new error related to this change:

<img width="1678" alt="Screen Shot 2021-08-01 at 11 35 09 AM" src="https://user-images.githubusercontent.com/534034/127778589-404d1cac-8623-49f4-b636-bf77420fafb0.png">

> Can we update the package to only allows React version >=17?

We currently support react@16.3 or greater. I do not want to change that (at least not in this PR). Also are you measuring bundle size using the dist files? We have react as a peer dependency so its not included in the actual bundles.